### ISSUE TITLE
Feature Request To Add google sign  plugin for macos and windows app

### ISSUE THREAD

**Issue by alihassan143** [flutter/flutter#103682]

**Comment by darshankawar** [comment]
There's already an open request for macOS : https://github.com/flutter/flutter/issues/46157, but I don't see the same for Windows yet, so updating issue title to reflect the same.

**Comment by a8775** [comment]
Does exist a ready to use workaround eg. by execuing flutter_webview_desktop or something other?

**Comment by alihassan143** [comment]
yes exists but it not working in the ways it should work most of the data is missing

**Comment by a8775** [comment]
Super, could you give a link? I searched but couldn't find.

**Comment by alihassan143** [comment]
@a8775  
https://pub.dev/packages/desktop_webview_auth
link

**Comment by dotjon0** [comment]
Any update to this feature request and whether or not support for Windows is planned?

**Comment by bselwe** [comment]
We run into some limitations with @rafaelortizzableh when using [desktop_webview_auth](https://pub.dev/packages/desktop_webview_auth), mainly, the user needs to log in to their social platform, even though they may already be logged in in their preferred (default) browser. This is because the webview acts as a separate, incognito browser window.

Here's the workaround we used to get the Google authentication working in Windows and MacOS using [`url_launcher`](https://pub.dev/packages/url_launcher) with [`app_links`](https://pub.dev/packages/app_links): [Social authentication in Flutter Desktop apps](https://bselwesiuk.medium.com/social-authentication-in-flutter-desktop-apps-c8c0599e13d3).

**Comment by dotjon0** [comment]
Any update on this for flutter native support via firebase auth?

**Comment by dotjon0** [comment]
Any update on this for flutter native support via firebase auth?

**Comment by tnc1997** [comment]
I would like to share a desktop implementation of `google_sign_in` that uses the [Authorization Code Flow with Proof Key for Code Exchange (PKCE)](https://datatracker.ietf.org/doc/html/rfc7636) flow, with the [loopback address](https://developers.google.com/identity/protocols/oauth2/native-app#redirect-uri_loopback) as the `redirect_uri`, to get tokens. I would be more than happy to work with the maintainers of `google_sign_in` to get a desktop implementation endorsed for Linux and Windows.

https://github.com/tnc1997/flutter-google-sign-in-desktop

**Comment by selvam920** [comment]
any update? can you guide me how google quick share windows app used google authentication?

**Comment by Patrick386** [comment]
Currently, the google_sign_in package does not support Windows, which is a critical feature for our production project. This functionality is essential for cross-platform development, especially as Windows remains a significant platform for desktop applications. The lack of support is causing substantial delays in our deployment timeline. Given the importance of this feature, I strongly urge the team to prioritize it and assign a 'P1' (Priority 1) label to ensure it receives the attention it urgently needs. 
Thank you for your consideration and hard work on this project.

**Comment by alihassan143** [comment]
> Currently, the google_sign_in package does not support Windows, which is a critical feature for our production project. This functionality is essential for cross-platform development, especially as Windows remains a significant platform for desktop applications. The lack of support is causing substantial delays in our deployment timeline. Given the importance of this feature, I strongly urge the team to prioritize it and assign a 'P1' (Priority 1) label to ensure it receives the attention it urgently needs. Thank you for your consideration and hard work on this project.

work around is that 
you can use deep linking using protcol handler to authenticate user using your website than redirecting user to the app like other apps do 
i did that in our company and its working fine without any issue

**Comment by tnc1997** [comment]
> I would be more than happy to work with the maintainers of `google_sign_in` to get a desktop implementation endorsed for Linux and Windows.

@stuartmorgan thank you very much for the work that you and the team have done on `google_sign_in`.

Is an implementation for Linux and Windows something that you and the team may be open to endorsing?

**Comment by stuartmorgan** [comment]
@dotjon0, @Patrick386, and anyone else considering commenting here: please read https://github.com/flutter/flutter/blob/master/docs/contributing/issue_hygiene/README.md#do-not-add-me-too-or-same-or-is-there-an-update-comments-to-issues-or-prs and https://github.com/flutter/flutter/blob/master/docs/contributing/issue_hygiene/README.md#escalating-an-issue-that-has-the-wrong-priority

**Comment by stuartmorgan** [comment]
@tnc1997 [These](https://github.com/flutter/flutter/issues/46157#issuecomment-562249211) [two](https://github.com/flutter/flutter/issues/46157#issuecomment-562742154) comments apply to Linux and Windows implementations as well (they used to apply to macOS, but then the SDK added macOS support).

Step 4 in the README of the implementation you linked to appears to be a manifestation of the kind of fundamental mismatch I was concerned about in those comments. It's certainly something you could publish for use as an unendorsed implementation, but if there isn't a way to resolve that mismatch it's probably not an approach we would use for an endorsed implementation.

**Comment by stuartmorgan** [comment]
@dotjon0 Please see https://github.com/flutter/flutter/blob/master/docs/contributing/issue_hygiene/README.md#when-will-my-bug-be-fixed for the answer to questions about when issues will be fixed.

> Visibility on this will help stop people asking every x months.

Our guidance about how to interact with issues is not conditional on having timelines for fixing every issue, especially since as noted in the guidance we don't provide timelines for most issues.

**Comment by stuartmorgan** [comment]
> raising the unanswered questions that we and others have raised in this ticket (which have been closed down, unsatisfactory or given generic responses which are of no value)
> [...]
> (1) will this be supported at some point

There is no answer other than the generic one. As it says in the docs I linked to (emphasis added):

>> If the issue is labeled with priorities P0 or P1, or if the issue is assigned, we are likely to address it in the near term; we just need to find time.
>>
>> **Otherwise, we don't know when we're going to fix it. We may never get to it.**

This issue is not P0 or P1, so the bolded text applies. If you find that answer unsatisfactory, the link I provided about escalation applies, as it says in the guidance.

> (2) if not is there anything blocking this issue

It is blocked on one of:
- Google Sign-In adding Windows support,
- A compatible implementation being possible with a different Google SDK, or
- `google_sign_in` going through a significant breaking change to make its API less closely tied to the Google Sign-In SDK (which is a possible outcome given that Google Sign In for Android and Google Sign In for Web are now both labeled as legacy, resulting in a more divergent auth ecosystem).

**Comment by vishnuagbly** [comment]
Hi Team, For a workaround, till we have official support, I have created a package called [google_sign_in_all_platforms](https://pub.dev/packages/google_sign_in_all_platforms), this uses web OAuth2.0 via the default browser for Desktop, like many desktop applications, to provide support for the windows platform. Will be happy to address any issues.
